Classes/Service/DemandService.php

```diff
@@ -187,6 +231,91 @@ public function getInnerBoundsForProperty(string $propertyName): array
      */
     protected function getConfiguredDemandProviders(): array
     {
-        // TODO: Initialize DemandProvider objects configured in TypoScript and return in order of execution
+        $configurationManager = GeneralUtility::makeInstance(ConfigurationManager::class);
+        $config = $configurationManager->getConfiguration(ConfigurationManager::CONFIGURATION_TYPE_FULL_TYPOSCRIPT)['config.']['tx_demander.'];
+        $demandProviders = [];
+
+        foreach ($config['demandProviders.'] as $id => $demandProvider)
+        {
+            $demandProviders[$id] = GeneralUtility::makeInstance($demandProvider);
+        }
+        return $demandProviders;
+    }
+
+    /**
+     * Transform necessary values from string to integer.
+     *
+     * @param array $restrictionsArray
+     * @return array
+     */
+    protected function restrictionsToInt(array $restrictionsArray): array
+    {
+        $restrictions = [];
+
+        foreach ($restrictionsArray as $key => $restriction) {
+            if ($restriction['value']) {
+                $value = (is_numeric($restriction['value'])) ? (int)$restriction['value'] : $restriction['value'];
+                $restriction = array_replace($restriction, ['value' => $value]);
+                $restrictions[$key] = $restriction;
+            }else{
+                $restrictions[$key] = $this->restrictionsToInt($restriction);
+            }
+        }
+
+        return $restrictions;
+    }
+
+    /**
+     * Resolve field name from alias or from array if OR, AND provided.
+     *
+     * @param mixed $alias
+     * @param null|string $rootKey
+     * @return mixed
+     */
+    protected function getFieldPropsFromAlias($alias, $rootKey = null)
+    {
+        if ($rootKey !== null){
+            $rootKey = mb_substr($rootKey, 0, -1);
+            $fieldProps = [];
+
+            foreach ($alias as $key => $subAlias){
+                $fieldProps[$rootKey][$this->getFieldPropsFromAlias($key)] = array(
+                    $subAlias
+                );
```

Comment:
No old `array()` calls, please. ;-)

```suggestion
                $fieldProps[$rootKey][$this->getFieldPropsFromAlias($key)] = [$subAlias];
```